Aws/Ec2/Client.py

```diff
@@ -28,3 +29,71 @@ def describe_regions(self) -> Optional[Region]:
         :return: Region object, or None if not found
         """
         return None
+
+    def run_instances(
+            self,
+            instance_type=None,
+            image_id=None,
+            user_data=None,
+            shutdown_behavior=None,
+            network_interfaces=None,
+            min_count=1,
+            max_count=1
+    ):
+        """
+        Start a new EC2 instance
+
+        :param instance_type: Optional Ec2 Instance type/size
+        :type instance_type: Optional[str]
+
+        :param image_id: Optional Instance AMI ID to run the instance with
+        :type image_id: Optional[str]
+
+        :param shutdown_behavior: Optional behaviour of the EC2 instance storage when shutdown
+        :type shutdown_behavior: Optional[str]
+
+        :param user_data: Optional Bash script to run on the instance at startup
+        :type user_data: Optional[str]
+
+        :param network_interfaces: Optional dict for the creation of a network interface to attach to the instance
+        :type network_interfaces: Optional[Dict]
+
+        :param min_count: Optional count for minimum instances to launch
+        :type min_count: Optional[int]
+
+        :param max_count: Optional count for maximum instances to launch
+        :type max_count: Optional[int]
+
```

Comment:
I could be reading this wrong- but is monitoring not required?

Monitoring (dict) --
Specifies whether detailed monitoring is enabled for the instance.

Enabled (boolean) -- [REQUIRED]
Indicates whether detailed monitoring is enabled. Otherwise, basic monitoring is enabled.

Reply:
Not from my experience, I believe that is saying the flag is required if the dict is defined?